packages/flame/lib/src/components/component.dart

```diff
@@ -44,6 +44,25 @@ class Component {
   Component? get parent => _parent;
   Component? _parent;
 
+  /// Returns the first child that matches the given type [T].
+  T? firstChild<T extends Component>() {
+    final match = children.whereType<T>();
+
+    final it = match.iterator;
+    return it.moveNext() ? it.current : null;
+  }
+
+  /// Returns the last child that matches the given type [T].
+  T? lastChild<T extends Component>() {
+    final match = children.whereType<T>();
```

Comment:
You can do `children.reversed` here to not have to go through the full list (since reversed is already cached).